translate/convert/xliff2po.py

```diff
@@ -67,11 +69,6 @@ def converttransunit(self, transunit):
 
     def convertstore(self, inputfile, duplicatestyle="msgctxt"):
         """Converts a .xliff file to .po format"""
-        # XXX: The inputfile is converted to string because Pootle supplies
-        # XXX: a PootleFile object as input which cannot be sent to PoXliffFile
-        # XXX: The better way would be to have a consistent conversion API.
-        if not isinstance(inputfile, (file, wStringIO.StringIO)):
-            inputfile = str(inputfile)
```

Comment:
Just did some bisect and `PootleFile` object for which this hack was introduced seems to be gone from Pootle since 2009.

So I say lets move this specific change to a separate commit with a detailed commit message including the relevant parts of this chat. That way if we have to investigate it will be easier to find a small change with a detailed description and that is easily revertable.